cdap-distributions/bin/build_artifact_upload.sh

```diff
@@ -83,22 +83,35 @@ function sync_build_artifacts_to_server () {
   do
     decho "PACKAGE=${i}"
     _package=`echo ${i} | awk -F / '{ print $(NF) }'`
-    if [[ "${_package}" == *rpm ]]
-    then
-      _version_stub=`echo ${_package} | awk -F - '{ print $(NF-1) }'`
-    else
-      if [[ "${_package}" == *_* ]]
-      then
+
+    case $_package in
+
+      cdap.*_[0-9]\.[0-9]\.[0-9]\.[0-9]+.*deb | cdap.*-[0-9]\.[0-9]\.[0-9]\.[0-9]+.*rpm )  ## snapshots with timestamp
+        _version_stub=`echo ${_package} | awk -F _ '{ print $2 }'`
+        ;;
+
+      *_all.deb )                             ## debian standard
+        _version_stub=`echo ${_package} | awk -F _ '{ print $2 }' | awk -F - '{ print $1 }'`
+        ;;
+
+      *noarch.rpm )                           ## rpm standard
+        _version_stub=`echo ${_package} | awk -F - '{ print $(NF-1) }'`
+        ;;
+
+      *_* )                                   ## default debian case
```

Comment:
if you want this to catch only debians then add a .deb to the end.  otherwise this is a catchall for anything with a _, including rpms